Could not read from file: /errori18n/node_modules/dist/vue-i18n.mjs

### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.18.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.19.2`
- Builder:          `vite`
- User Config:      `ssr`, `css`, `build`, `vite`, `modules`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.7`, `()`
- Build Modules:    `-`


### Reproduction

link to a repo:


[Reproduction repo](https://github.com/artmarydotir/nuxt-i18n-error)

### Describe the bug

When running: **npm run dev** error accrued.

```
node_modules/@nuxtjs/i18n/node_modules/vue-i18n-routing/node_modules/@intlify/vue-i18n-bridge/lib/index.mjs:10:14:
      10 │ export * from 'vue-i18n'
         ╵               ~~~~~~~~~
 ```
![Screenshot from 2022-12-20 22-25-19](https://user-images.githubusercontent.com/11363931/208744489-3109a4b6-b94e-490e-bda3-6e0ce6b00e6b.png)


### Additional context

_No response_

### Logs

_No response_

I have the same issue with the error:

[ERROR] Could not read from file: ../nuxt-loco/playground/node_modules/dist/vue-i18n.mjs

I assume this is due to the architecture of the projet as playground is inside another folder. relative access path ? 
This is working if i do `npx nuxi dev` inside the playground folder.

I need to use node 16 because of my hosting.
------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

Here the reproduction link in the playground : https://github.com/gaetansenn/nuxt-loco

--- a/src/alias.ts
+++ b/src/alias.ts
@@ -1,11 +1,11 @@
 import createDebug from 'debug'
-import { resolvePath } from '@nuxt/kit'
 import { resolve } from 'pathe'
 import {
   VUE_I18N_PKG,
   VUE_I18N_BRIDGE_PKG,
   VUE_ROUTER_BRIDGE_PKG,
   VUE_I18N_ROUTING_PKG,
+  SHARED_PKG,
   MESSAGE_COMPILER_PKG
 } from './constants'
 import { pkgModulesDir } from './dirs'
@@ -26,13 +26,13 @@ export async function setupAlias(nuxt: Nuxt) {
   debug('vue-i18n alias', nuxt.options.alias[VUE_I18N_PKG])
 
   // resolve @intlify/shared
-  nuxt.options.alias['@intlify/shared'] = await resolvePath('@intlify/shared')
-  nuxt.options.build.transpile.push('@intlify/shared')
-  debug('@intlify/shared alias', nuxt.options.alias['@intlify/shared'])
+  nuxt.options.alias[SHARED_PKG] = await resolveSharedAlias(pkgModulesDir, nuxt, pkgMgr)
+  nuxt.options.build.transpile.push(SHARED_PKG)
+  debug('@intlify/shared alias', nuxt.options.alias[SHARED_PKG])
 
   nuxt.options.alias['@intlify/message-compiler'] = await resolveMessageCompilerAlias(pkgModulesDir, nuxt, pkgMgr)
-  nuxt.options.build.transpile.push('@intlify/message-compiler')
-  debug('@intlify/message-compiler alias', nuxt.options.alias['@intlify/message-compiler'])
+  nuxt.options.build.transpile.push(MESSAGE_COMPILER_PKG)
+  debug('@intlify/message-compiler alias', nuxt.options.alias[MESSAGE_COMPILER_PKG])
 
   // resolve @intlify/vue-router-bridge
   nuxt.options.alias[VUE_ROUTER_BRIDGE_PKG] = await resolveVueRouterBridgeAlias(pkgModulesDir, nuxt, pkgMgr)
@@ -52,10 +52,12 @@ export async function setupAlias(nuxt: Nuxt) {
 
 /**
  * NOTE:
- *  The following packages may not be able to resolve the file paths of the target ES modules with `resolvePath`
- *  so they resolve them on their own (sometimes, these are resolved as `cjs`)
+ *  The following packages maybe installed in directories
+ *  where the package manager `node_modules` installation algorithm cannot resolve the file path of the target ES module with `resolvePath` (`@nuxt/kit`).
+ *  (e.g. npm peerDependencies).
  *  - `vue-i18n`
  *  - `vue-i18n-routing`
+ *  - `@intlify/shared`
  *  - `@intlify/vue-i18n-bridge`
  *  - `@intlify/vue-router-bridge`
  */
@@ -69,10 +71,10 @@ export async function resolveVueI18nAlias(pkgModulesDir: string, nuxt: Nuxt, pkg
   const targets = [
     // for Nuxt layer
     ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
-    // 1st, try to resolve from `node_modules` (hoisted case)
-    resolve(rootDir, 'node_modules', modulePath),
-    // 2nd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
+    // try to resolve from `node_modules/@nuxtjs/i18n/node_modules` (not hoisted case)
     resolve(pkgModulesDir, modulePath),
+    // try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
     // workspace directories
     resolve(workspaceDir, 'node_modules', modulePath)
   ]
@@ -81,24 +83,68 @@ export async function resolveVueI18nAlias(pkgModulesDir: string, nuxt: Nuxt, pkg
   return tryResolve(VUE_I18N_PKG, targets, pkgMgr)
 }
 
+export async function resolveSharedAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
+  const { rootDir, workspaceDir } = nuxt.options
+  const modulePath = `${SHARED_PKG}/dist/shared.mjs` as const
+  const targets = [
+    // for Nuxt layer
+    ...getLayerRootDirs(nuxt).map(root => resolve(root, `${VUE_I18N_PKG}/node_modules`, modulePath)),
+    ...getLayerRootDirs(nuxt).map(root => resolve(root, `${MESSAGE_COMPILER_PKG}/node_modules`, modulePath)),
+    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
+    // try to resolve from `node_modules/@nuxtjs/i18n/node_modules` (not hoisted case)
+    resolve(pkgModulesDir, modulePath),
+    // try to resolve from `node_modules/@nuxtjs/i18n/node_modules/vue-i18n/node_modules` (not hoisted case)
+    resolve(pkgModulesDir, `${VUE_I18N_PKG}/node_modules`, modulePath),
+    // try to resolve from `node_modules/@nuxtjs/i18n/node_modules/@intlify/message-compiler/node_modules` (not hoisted case)
+    resolve(pkgModulesDir, `${MESSAGE_COMPILER_PKG}/node_modules`, modulePath),
+    // try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
+    // workspace directories
+    resolve(workspaceDir, 'node_modules', `${VUE_I18N_PKG}/node_modules`, modulePath),
+    resolve(workspaceDir, 'node_modules', `${MESSAGE_COMPILER_PKG}/node_modules`, modulePath),
+    resolve(workspaceDir, 'node_modules', modulePath)
+  ]
+  debug(`${SHARED_PKG} resolving from ...`, targets)
+
+  return tryResolve(SHARED_PKG, targets, pkgMgr)
+}
+
+export async function resolveMessageCompilerAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
+  const { rootDir, workspaceDir } = nuxt.options
+  const modulePath = `${MESSAGE_COMPILER_PKG}/dist/message-compiler.mjs` as const
+  const targets = [
+    // for Nuxt layer
+    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
+    // try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
+    resolve(pkgModulesDir, modulePath),
+    // try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
+    // workspace directories
+    resolve(workspaceDir, 'node_modules', modulePath)
+  ]
+  debug(`${MESSAGE_COMPILER_PKG} resolving from ...`, targets)
+
+  return tryResolve(MESSAGE_COMPILER_PKG, targets, pkgMgr)
+}
+
 async function resolveVueI18nBridgeAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
   const { rootDir, workspaceDir } = nuxt.options
   const modulePath = `${VUE_I18N_BRIDGE_PKG}/lib/index.mjs` as const
   const targets = [
     // for Nuxt layer
-    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
     ...getLayerRootDirs(nuxt).map(root => resolve(root, `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath)),
-    // 1st, try to resolve from `node_modules` (hoisted case)
-    resolve(rootDir, 'node_modules', modulePath),
-    // 2nd, try to resolve from `node_modules/vue-i18n-routing` (not hoisted case)
-    resolve(rootDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
-    // 3rd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
+    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
+    // try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
     resolve(pkgModulesDir, modulePath),
-    // 4th, try to resolve from `node_modules/@nuxtjs/i18n/node_modules/vue-i18n-routing` (not hoisted case)
+    // try to resolve from `node_modules/@nuxtjs/i18n/node_modules/vue-i18n-routing` (not hoisted case)
     resolve(pkgModulesDir, `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
+    // try to resolve from `node_modules/vue-i18n-routing` (hoisted case)
+    resolve(rootDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
+    // try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
     // workspace directories
-    resolve(workspaceDir, 'node_modules', modulePath),
-    resolve(workspaceDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath)
+    resolve(workspaceDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
+    resolve(workspaceDir, 'node_modules', modulePath)
   ]
   debug(`${VUE_I18N_BRIDGE_PKG} resolving from ...`, targets)
 
@@ -110,19 +156,19 @@ async function resolveVueRouterBridgeAlias(pkgModulesDir: string, nuxt: Nuxt, pk
   const modulePath = `${VUE_ROUTER_BRIDGE_PKG}/lib/index.mjs` as const
   const targets = [
     // for Nuxt layer
-    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
     ...getLayerRootDirs(nuxt).map(root => resolve(root, `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath)),
-    // 1st, try to resolve from `node_modules` (hoisted case)
-    resolve(rootDir, 'node_modules', modulePath),
-    // 2nd, try to resolve from `node_modules/vue-i18n-routing` (not hoisted case)
-    resolve(rootDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
-    // 3rd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
+    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
+    // try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
     resolve(pkgModulesDir, modulePath),
-    // 4th, try to resolve from `node_modules/@nuxtjs/i18n/node_modules/vue-i18n-routing` (not hoisted case)
+    // try to resolve from `node_modules/@nuxtjs/i18n/node_modules/vue-i18n-routing` (not hoisted case)
     resolve(pkgModulesDir, `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
+    // try to resolve from `node_modules/vue-i18n-routing` (hoisted case)
+    resolve(rootDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
+    // try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
     // workspace directories
-    resolve(workspaceDir, 'node_modules', modulePath),
-    resolve(workspaceDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath)
+    resolve(workspaceDir, 'node_modules', `${VUE_I18N_ROUTING_PKG}/node_modules`, modulePath),
+    resolve(workspaceDir, 'node_modules', modulePath)
   ]
   debug(`${VUE_ROUTER_BRIDGE_PKG} resolving from ...`, targets)
 
@@ -135,32 +181,14 @@ export async function resolveVueI18nRoutingAlias(pkgModulesDir: string, nuxt: Nu
   const targets = [
     // for Nuxt layer
     ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
-    // 1st, try to resolve from `node_modules` (hoisted case)
-    resolve(rootDir, 'node_modules', modulePath),
-    // 2nd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
+    // try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
     resolve(pkgModulesDir, modulePath),
+    // try to resolve from `node_modules` (hoisted case)
+    resolve(rootDir, 'node_modules', modulePath),
     // workspace directories
     resolve(workspaceDir, 'node_modules', modulePath)
   ]
   debug(`${VUE_I18N_ROUTING_PKG} resolving from ...`, targets)
 
   return tryResolve(VUE_I18N_ROUTING_PKG, targets, pkgMgr)
-}
-
-export async function resolveMessageCompilerAlias(pkgModulesDir: string, nuxt: Nuxt, pkgMgr: PackageManager) {
-  const { rootDir, workspaceDir } = nuxt.options
-  const modulePath = `${MESSAGE_COMPILER_PKG}/dist/message-compiler.mjs` as const
-  const targets = [
-    // for Nuxt layer
-    ...getLayerRootDirs(nuxt).map(root => resolve(root, 'node_modules', modulePath)),
-    // 1st, try to resolve from `node_modules` (hoisted case)
-    resolve(rootDir, 'node_modules', modulePath),
-    // 2nd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
-    resolve(pkgModulesDir, modulePath),
-    // workspace directories
-    resolve(workspaceDir, 'node_modules', modulePath)
-  ]
-  debug(`${MESSAGE_COMPILER_PKG} resolving from ...`, targets)
-
-  return tryResolve(MESSAGE_COMPILER_PKG, targets, pkgMgr)
 }

--- a/src/constants.ts
+++ b/src/constants.ts
@@ -2,10 +2,11 @@ import type { NuxtI18nOptionsDefault } from './options'
 
 export const NUXT_I18N_MODULE_ID = '@nuxtjs/i18n' as const
 export const VUE_I18N_PKG = 'vue-i18n' as const
+export const SHARED_PKG = '@intlify/shared' as const
+export const MESSAGE_COMPILER_PKG = '@intlify/message-compiler' as const
 export const VUE_I18N_BRIDGE_PKG = '@intlify/vue-i18n-bridge' as const
 export const VUE_ROUTER_BRIDGE_PKG = '@intlify/vue-router-bridge' as const
 export const VUE_I18N_ROUTING_PKG = 'vue-i18n-routing' as const
-export const MESSAGE_COMPILER_PKG = '@intlify/message-compiler' as const
 
 // Options
 const STRATEGY_PREFIX = 'prefix'

--- a/src/dirs.ts
+++ b/src/dirs.ts
@@ -1,8 +1,6 @@
 import createDebug from 'debug'
 import { fileURLToPath } from 'node:url'
-import { dirname, resolve, parse as parsePath } from 'pathe'
-import { VUE_I18N_ROUTING_PKG } from './constants'
-import { getPackageManagerType, tryResolve } from './utils'
+import { dirname, resolve } from 'pathe'
 
 const debug = createDebug('@nuxtjs/i18n:dirs')
 
@@ -15,19 +13,3 @@ debug('distDir', distDir)
 debug('runtimeDir', runtimeDir)
 debug('pkgDir', pkgDir)
 debug('pkgModulesDir', pkgModulesDir)
-
-export async function resolveVueI18nRoutingDtsPath(id: string, rootDir: string) {
-  const pkgMgr = await getPackageManagerType()
-  const dtsPath = `${VUE_I18N_ROUTING_PKG}/dist/${id}`
-  const targets = [
-    // 1st, try to resolve from `node_modules` (hoisted case)
-    resolve(rootDir, 'node_modules', dtsPath),
-    // 2nd, try to resolve from `node_modules/@nuxtjs/i18n` (not hoisted case)
-    resolve(pkgModulesDir, dtsPath)
-  ]
-  debug(`${VUE_I18N_ROUTING_PKG} resolving from ...`, targets)
-
-  const resolved = await tryResolve(VUE_I18N_ROUTING_PKG, targets, pkgMgr, '.d.ts')
-  const parsed = parsePath(resolved)
-  return `${parsed.dir}/${parsed.name}`
-}

